src/app.rs

```diff
@@ -176,73 +161,64 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
                 .unwrap_or_default(),
         );
 
-        let app_init_cfg = AppInitCfg {
-            mount_type: MountType::Takeover,
-            phantom: PhantomData,
-            into_after_mount: Box::new({
-                let base_path = Rc::clone(&base_path);
-                move |url: Url,
-                      orders: &mut OrdersContainer<Ms, Mdl, INodes, GMs>|
-                      -> AfterMount<Mdl> {
-                    let url = url.skip_base_path(&base_path);
-                    let model = init(url, orders);
-                    AfterMount::new(model).url_handling(UrlHandling::None)
-                }
-            }) as Box<dyn IntoAfterMount<Ms, Mdl, INodes, GMs>>,
-        };
-        let app = Self::new(
-            update,
-            None,
-            view,
-            root_element,
-            None,
-            None,
-            Some(app_init_cfg),
-            base_path,
+        let root_element = root_element.get_element().expect("get root element");
+        let app = Self::new(update, view, root_element, base_path.clone());
```

Comment:
Please use `Rc::clone` where possible (see the old code in this diff).

https://doc.rust-lang.org/book/ch15-04-rc.html#using-rct-to-share-data:
> We could have called a.clone() rather than Rc::clone(&a), but Rust’s convention is to use Rc::clone in this case. The implementation of Rc::clone doesn’t make a deep copy of all the data like most types’ implementations of clone do. The call to Rc::clone only increments the reference count, which doesn’t take much time. Deep copies of data can take a lot of time. By using Rc::clone for reference counting, we can visually distinguish between the deep-copy kinds of clones and the kinds of clones that increase the reference count. When looking for performance problems in the code, we only need to consider the deep-copy clones and can disregard calls to Rc::clone.

Reply:
base_path is already an Rc. Are you talking about the app itself?

That sould work for the app, but I think it might be digging ourselves into a hole. At the moment, making a new app is dependent on having a model, but once you have that app, it needs to take ownership of the model. `Rc` can help us get around it, but it feels a bit hacky - kinda like a circular dependency. I suspect that there's a better way to do it.

Outside of this piece of code, the nature of the data doesn't seem to create any need of a ref-count. I think part of cleaning up `start` should include clearing away deep-copies and unnecisary `Rc` wrapping. 

How about we pull this into a development branch? That way we can have a go at not having deep-copy AND not wrapping it in `Rc`. If it doesn't work out, we can fix it up before bringing it into production.

Reply:
> base_path is already an Rc. Are you talking about the app itself?

Please read my comment and copy-pasted explanation again :)

`Rc::clone(&base_path)` vs `base_path.clone()` => just replace `.clone` with `Rc::clone`.